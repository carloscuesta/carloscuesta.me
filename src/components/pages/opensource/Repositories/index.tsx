import { useState } from 'react'

import SectionTitle from 'src/components/shared/SectionTitle'
import { type Repository } from 'src/utils/api/github/mutators'

type Props = { repositories: Array<Repository> }

const Repositories = (props: Props) => {
  const [search, setSearch] = useState<string>('')
  const repositories = props.repositories.filter(({ name, description }) => {
    if (!search) return true

    return name.toLowerCase().includes(search.toLowerCase()) || 
      description?.toLowerCase().includes(search.toLowerCase())
  })

  return (
    <section>
      <SectionTitle title='Repositories' />
      <input type='search' className='outline-2 outline-offset-2 outline-pink-200 border-2 border-pink-300 w-full mb-4 p-2 rounded-md dark:bg-neutral-900 dark:text-neutral-100' placeholder='Search repositories...' onChange={(event)  => setSearch(event.target.value)} />
      <ul>
        {repositories.length
          ? repositories.map((repository) => (
            <li
              className='mb-4 dark:bg-neutral-900 dark:hover:bg-neutral-800 dark:transition-colors rounded-md overflow-hidden'
              key={repository.name}
            >
              <a className={`flex p-4 flex-col md:flex-row border-l-4 language-${repository.language?.toLowerCase()}`}
                href={repository.url}
                target='_blank'
                rel='noopener noreferrer'
              >
                <b className='font-semibold'>{repository.name}</b>
                <p className='flex-1 my-2 md:mx-4 md:my-0 opacity-75'>{repository.description}</p>
                <div className='flex items-center text-sm'>
                  <svg className='fill-amber-400 mx-1 p-[1px]' viewBox='0 0 16 16' width='16' height='16' fill='currentColor'>
                    <path d='M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25z' />
                  </svg>
                  <span>{repository.stars.format}</span>
                </div>
              </a>
            </li>
          ))
          : <p>No repositories found.</p>
        }
      </ul>
    </section>
  )
}

export default Repositories
