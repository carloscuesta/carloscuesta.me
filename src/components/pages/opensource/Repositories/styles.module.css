.list {
  list-style-type: none;
  padding: 0;
}

.list li {
  border-left: .20rem solid;
  border-left-color: var(--backgroundSurface);
  border-radius: .20rem;
  box-shadow: 0 1px 4px var(--shadowColor);
  overflow: hidden;
  transition: box-shadow 0.2s ease;
  margin: 1rem 0;
}

.list li:hover {
  box-shadow: 0 .25rem .5rem var(--shadowColor);
}

.list a:focus {
  outline-offset: -.05rem;
}

.description {
  flex: 1;
  margin: 0;
}

.name {
  margin-right: 1rem;
}

.list a {
  display: flex;
  padding: 1rem;
  text-decoration: none;
  color: var(--text);
}

.starsContainer {
  display: flex;
  align-items: center;
}

.stars {
  margin-right: .25rem;
  color: var(--languageJavaScript);
}

.list li:global(.javascript) {
  border-left-color: var(--languageJavaScript);
}

.list li:global(.typescript) {
  border-left-color: var(--languageTypeScript);
}

.list li:global(.shell) {
  border-left-color: var(--languageShell);
}

.list li:global(.css) {
  border-left-color: var(--languageCss);
}

.list li:global(.html) {
  border-left-color: var(--languageHtml);
}

@media (prefers-color-scheme: dark) {
  .list li {
    background-color: var(--backgroundSurface);
    box-shadow: none;
    transition: background 0.2s ease-out;
  }

  .list li:hover {
    background-color: var(--backgroundElevation);
    box-shadow: none;
  }
}

@media (max-width: 768px) {
  .list a {
    flex-direction: column;
  }

  .description {
    padding: .5rem 0;
  }
}
