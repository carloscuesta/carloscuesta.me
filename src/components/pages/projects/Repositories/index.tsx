import { useState } from 'react'

import SectionTitle from 'src/components/shared/SectionTitle'
import { type Repository } from 'src/utils/api/github/mutators'

type Props = { repositories: Array<Repository> }

const Repositories = (props: Props) => {
  const [search, setSearch] = useState<string>('')
  const repositories = props.repositories.filter(({ name, description }) => {
    if (!search) return true

    return (
      name.toLowerCase().includes(search.toLowerCase()) ||
      description?.toLowerCase().includes(search.toLowerCase())
    )
  })

  return (
    <section>
      <SectionTitle title="Repositories" />
      <input
        type="search"
        className="mb-4 w-full rounded-md border border-neutral-100 p-2 outline-2 outline-offset-2 outline-neutral-100 dark:border-neutral-800 dark:bg-neutral-900 dark:text-neutral-100 dark:outline-neutral-700"
        placeholder="Search repositories..."
        onChange={(event) => setSearch(event.target.value)}
      />
      <ul>
        {repositories.length ? (
          repositories.map((repository) => (
            <li key={repository.name}>
              <a
                href={repository.url}
                className="mb-2 grid grid gap-2 gap-2.5 rounded-lg rounded-md border border-neutral-100 p-3 p-4 transition-colors hover:border-neutral-300 dark:border-neutral-800 dark:hover:border-neutral-600"
              >
                <p className="font-semibold">{repository.name}</p>

                <div className="flex flex-col justify-between gap-4 sm:flex-row">
                  <p className="text-sm opacity-70">{repository.description}</p>

                  <div className="flex gap-2 text-sm">
                    {repository.language && (
                      <div className="flex items-center gap-1.5">
                        <div
                          className={`h-2 w-2 rounded-full language-project-${repository.language.toLowerCase()}`}
                        />
                        <span className="opacity-70">
                          {repository.language}
                        </span>
                      </div>
                    )}

                    <div className="flex items-center gap-1">
                      <svg
                        className="p-[1.5px] opacity-70"
                        viewBox="0 0 16 16"
                        width="16"
                        height="16"
                        fill="currentColor"
                      >
                        <path d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25z" />
                      </svg>
                      <span className="opacity-70">
                        {repository.stars.format}
                      </span>
                    </div>
                  </div>
                </div>
              </a>
            </li>
          ))
        ) : (
          <p>No repositories found.</p>
        )}
      </ul>
    </section>
  )
}

export default Repositories
