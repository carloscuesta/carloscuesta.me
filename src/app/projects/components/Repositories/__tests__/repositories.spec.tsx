import { render } from '@testing-library/react'

import Repositories from '../index'
import * as stubs from './stubs'

describe('Repositories', () => {
  it('should match Repositories component', () => {
    const { container } = render(<Repositories {...stubs.props} />)

    expect(container).toMatchSnapshot()
  })
})
