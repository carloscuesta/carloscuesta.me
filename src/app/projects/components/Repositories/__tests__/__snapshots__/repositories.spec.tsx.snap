// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories should match Repositories component 1`] = `
<div>
  <section>
    <h2
      class="my-6 block text-lg font-bold"
    >
      Repositories
    </h2>
    <input
      class="mb-4 w-full rounded-md border border-neutral-100 p-2 outline-2 outline-offset-2 outline-neutral-100 dark:border-neutral-800 dark:bg-neutral-900 dark:text-neutral-100 dark:outline-neutral-700"
      placeholder="Search repositories..."
      type="search"
    />
    <ul>
      <li>
        <a
          class="mb-2 grid gap-2.5 rounded-md border border-neutral-100 p-4 transition-colors hover:border-neutral-300 dark:border-neutral-800 dark:hover:border-neutral-600"
          href="https://github.com/carloscuesta/gitmoji"
        >
          <p
            class="font-semibold"
          >
            gitmoji
          </p>
          <div
            class="flex flex-col justify-between gap-4 sm:flex-row"
          >
            <p
              class="text-sm opacity-70"
            >
              An emoji guide for your commit messages. ðŸ˜œ 
            </p>
            <div
              class="flex gap-2 text-sm"
            >
              <div
                class="flex items-center gap-1.5"
              >
                <div
                  class="h-2 w-2 rounded-full language-project-javascript"
                />
                <span
                  class="opacity-70"
                >
                  javascript
                </span>
              </div>
              <div
                class="flex items-center gap-1"
              >
                <svg
                  class="p-[1.5px] opacity-70"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25z"
                  />
                </svg>
                <span
                  class="opacity-70"
                >
                  7,323
                </span>
              </div>
            </div>
          </div>
        </a>
      </li>
      <li>
        <a
          class="mb-2 grid gap-2.5 rounded-md border border-neutral-100 p-4 transition-colors hover:border-neutral-300 dark:border-neutral-800 dark:hover:border-neutral-600"
          href="https://github.com/carloscuesta/gitmoji-cli"
        >
          <p
            class="font-semibold"
          >
            gitmoji-cli
          </p>
          <div
            class="flex flex-col justify-between gap-4 sm:flex-row"
          >
            <p
              class="text-sm opacity-70"
            >
              A gitmoji interactive command line tool for using emojis on commits. ðŸ’»
            </p>
            <div
              class="flex gap-2 text-sm"
            >
              <div
                class="flex items-center gap-1.5"
              >
                <div
                  class="h-2 w-2 rounded-full language-project-javascript"
                />
                <span
                  class="opacity-70"
                >
                  javascript
                </span>
              </div>
              <div
                class="flex items-center gap-1"
              >
                <svg
                  class="p-[1.5px] opacity-70"
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25z"
                  />
                </svg>
                <span
                  class="opacity-70"
                >
                  2,400
                </span>
              </div>
            </div>
          </div>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
